### ISSUE TITLE
volatility: metadata column names containing periods cannot be interpreted

### ISSUE THREAD

**Issue by nbokulich** [qiime2/q2-longitudinal#133]
**Bug Description**
If metadata column names contain periods, that column cannot be viewed on the volatility plot. It seems that the periods may be interpreted as special characters.

**Steps to reproduce the behavior**
see [here](https://view.qiime2.org/visualization/?type=html&src=https%3A%2F%2Fforum-qiime2-org.s3.dualstack.us-west-2.amazonaws.com%2Foriginal%2F2X%2F7%2F7e2e3ff647baa4762c2dfd7586408ab1ada1025a.qzv) (sourced from forum xref below)

**Expected behavior**
Those columns should be viewable in the volatility plot.

**References**
[forum xref](https://forum.qiime2.org/t/issues-with-plotting-volatility/7241/4?u=nicholas_bokulich)

**Comment by thermokarst** [comment]
My guess is that some vega lookup is using the Javascript dot accessor (`data.peanut`), when it should be using the square bracket lookup (`data['peanut']`). The problem here is when you have a period in the key: `data.peanut.butter` (`{data: { peanut: { butter: 'yummy!' } } }`) vs `data['peanut.butter']` (`{data: { 'peanut.butter': 'peanut butter jelly time!' } }`).

**Comment by David-Rod** [comment]
Hey there @nbokulich and @thermokarst,

After reviewing this issue with @ebolyen and @Oddant1, it seems that a newer version of Vega has solved the issue. The Vega version used by the user was version `3.3.1`. The current version in the QIIME 2 source code is version `4.4.0`. I replicated this issue (as closely as I could) with current files from the tutorial, and Q2 View has no issue displaying the plot for group column names containing a`.`. 

TLDR: I think we can close this issue.

Thanks.

**Comment by nbokulich** [comment]
thanks for investigating @David-Rod !

Maybe we should a) pin a minimum vega version (if it is not already) and b) maybe also toss in a quick unit test? (though if we are positive this is related to the vega version then no need for a test). Interested to hear @thermokarst 's thoughts.
